.github/workflows/coverage.yaml

```diff
@@ -0,0 +1,51 @@
+name: Upload Coverage
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Repository
+        uses: actions/checkout@v2
+
+      - name: List Files
+        run: ls -R
+
+      - name: Setup Dart SDK
+        uses: dart-lang/setup-dart@v1.6.0
+        with:
+          sdk: "stable"
+      - name: Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: "3.16.0"
+
+      - name: Install Melos
+        run: dart pub global activate melos
+
+      - name: Install dependencies
+        run: melos bootstrap
+
+      - name: Run unit tests
+        run: melos run unit_test
+
+      - name: Create coverage directory
+        run: mkdir -p coverage
+
+      - name: Merge coverage
+        run: |
+          melos exec rm -rf coverage
+          melos run unit_test
+          melos exec --dir-exists="coverage" -- bash -c 'dart pub global run coverde filter --input coverage/lcov.info --output coverage/filtered.lcov.info --filters .g.dart'
```

Comment:
please split running the tests and calculating the coverage into 2 different steps. the coverage step should check if the tests passed successfully to run.